src/Admin/AdminHelper.php

```diff
@@ -148,14 +148,14 @@ public function appendFormFieldElement(AdminInterface $admin, $subject, $element
 
             if ($collection instanceof DoctrinePersistentCollection || $collection instanceof PersistentCollection) {
                 //since doctrine 2.4
-                $entityClassName = $collection->getTypeClass()->getName();
+                $modelClassName = $collection->getTypeClass()->getName();
             } elseif ($collection instanceof Collection) {
-                $entityClassName = $this->getEntityClassName($admin, explode('.', preg_replace('#\[\d*?\]#', '', $path)));
+                $modelClassName = $this->getModelClassName($admin, explode('.', preg_replace('#\[\d*?\]#', '', $path)));
```

Comment:
This class is not `final`. If `getEntityClassName` is overriden, it will no longer be called, and this is a BC break

Reply:
You're totally right.
I've restored the calls to these methods. Please, let me know if you have any other concern.